specification/azuredatatransfer/resource-manager/Microsoft.AzureDataTransfer/preview/2023-10-11-preview/azuredatatransfer.json

```diff
@@ -0,0 +1,2326 @@
+          }
+        },
+        "operationId": "Pipelines_ListBySubscription",
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Listed all pipelines under the subscription.",
+            "schema": {
+              "$ref": "#/definitions/pipelinesListResult"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "../../../../../common-types/resource-management/v3/types.json#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        }
+      }
+    },
+    "/providers/Microsoft.AzureDataTransfer/operations": {
+      "get": {
+        "tags": [
+          "Operations"
+        ],
+        "operationId": "Operations_List",
+        "x-ms-examples": {
+          "List operations": {
+            "$ref": "./examples/listOperations.json"
+          }
+        },
+        "description": "Lists all of the available Microsoft.AzureDataTransfer REST API operations.",
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Listed all of the available operations.",
+            "schema": {
+              "$ref": "../../../../../common-types/resource-management/v3/types.json#/definitions/OperationListResult"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "../../../../../common-types/resource-management/v3/types.json#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        }
+      }
+    }
+  },
+  "definitions": {
+    "actionRequest": {
+      "type": "object",
+      "description": "An action request",
+      "properties": {
+        "name": {
+          "type": "string",
+          "description": "Name of the action"
+        },
+        "payload": {
+          "type": "object",
+          "description": "Action request payload",
+          "additionalProperties": {}
+        }
+      }
+    },
+    "actionResult": {
+      "type": "object",
+      "description": "An action result",
+      "properties": {
+        "result": {
+          "type": "object",
+          "description": "Action request result",
+          "additionalProperties": {}
+        }
```

Comment:
and what would be in this?

Reply:
The payload for the request/response would need to be dynamic to support the open nature of the post action.

Reply:
This an anti-pattern in ARM APIs, as customers won't know what to provide in request payload and there is no client side (SDK side) validation that they can use to build their payload. That's why using additional properties is not allowed. Please sign up for ARM API Modeling Office Hours to discuss your use case: [https://armwiki.azurewebsites.net/introduction/office_hours.html](https://armwiki.azurewebsites.net/introduction/office_hours.html).

Reply:
The additional properties for the provider-level action have been removed.